internal/errlog/errlog.go

```diff
@@ -0,0 +1,62 @@
+package errlog
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+type forwarder struct{}
+
+func (f *forwarder) Write(p []byte) (n int, err error) {
+	Printf("%s", string(p))
+	return len(p), nil
+}
+
+type errlog struct {
+	log         *log.Logger
+	everWritten bool
+	permLogs    strings.Builder
+	prereqLogs  strings.Builder
+}
+
+var singletone errlog
+
+func init() {
+	singletone.log = log.New(os.Stderr, "", log.LstdFlags)
+	log.SetOutput(&forwarder{})
+
+	PermIssueF("Missing permissions errors:\n")
+	PrereqIssueF("Unmet Prerequisites errors:\n")
+}
+
+func SetOutput(writer io.Writer) {
+	singletone.log.SetOutput(writer)
+}
+
+func Printf(format string, args ...interface{}) {
+	singletone.everWritten = true
+	singletone.log.Printf(format, args...)
+}
+
+func PermIssueF(format string, args ...interface{}) {
+	singletone.permLogs.WriteString(fmt.Sprintf(format, args...))
+}
+
+func PrereqIssueF(format string, args ...interface{}) {
+	singletone.prereqLogs.WriteString(fmt.Sprintf(format, args...))
+}
+
+func FlushAll() {
+	singletone.log.Printf(singletone.permLogs.String())
```

Comment:
why not use strings.Builder for the error log as well

Reply:
0. It has no advantage that I can think of (do you have any?)
2. logs should be written on the fly
3. logs won't be written in case of panic (unless we add a defer at the beginning, making things more complicated). 
4. we want separate records for different prints

Reply:
0. it's more aligned with the other methods in the file & theoretically, you can have other goroutines writing to the error log while your flushing the string builders, thus having an inconsistent output
1. agree
2. true for the strings.builders as well
4. what do you mean

Reply:
let's continue in slack